### DISCUSSION TITLE
LDAP search by filter not only base

### DISCUSSION THREAD

**Discussion by dorianim** [directus/directus#9475] [Feature Requests]
Hi,

Currently, the LDAP implementation only allows users and groups from one base DN. This is a problem for some setups, as all groups and all users are not necessarily stored in only one DN. 
I think, this should be quite easy to implement. We would just have to make it possible for users to specify a custom group and user filter and to change the search scope from `one` to `sub`. I think this wouldn't come with any downsides. 
What do you think @aidenfoxx?

Regards,
Dorian

**Comment by aidenfoxx** [comment]
The only downside is the cost of processing. Obviously it's more costly to search sub-directores. We could add a config variable to set the search scope. As you said, that is the only thing that needs changing.

I used "one" as the default because it was the most appropriate for my use case.

EDIT: I also realize that this could increase the chance for conflicts if you have sub-directories containing users with the same attribute e.g. "Developers -> John" and "Agents -> John". A sub-search would return both, and we'd use the first. This can also occur on the "one" level, but if you set up your server like that then you deserve problems! Either way, I feel enabling "sub" would be a "use at your own risk" kinda thing.

**Reply by dorianim** [reply]
Alight, that makes sense. We could default to the current configuration. This would mean no performance reduction by default, and only for the users who need it (like me).

**Reply by MiniDigger** [reply]
Since I ran into this today to, just stating here that I have a usecase for sub on group search, because we bundle groups in "folders"

**Reply by dorianim** [reply]
I'll take a look at this and create a PR once I have some time.

**Reply by aidenfoxx** [reply]
@dorianim Too slow! #9529 😁

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
